src/Form/Form.js

```diff
@@ -157,22 +156,22 @@ class Form {
      * @param {KeyboardEvent | null} [e]
      */
     hasFocus (e) {
-        return this.form.contains(document.activeElement) || this.form.contains(/** @type HTMLElement */(e?.target))
+        return this.form.contains(getActiveElement()) || this.form.contains(/** @type HTMLElement */(e?.target))
```

Comment:
This new function recurses within shadow trees in search for the `activeElement`.